koku/masu/processor/tasks.py

```diff
@@ -915,3 +920,54 @@ def remove_stale_tenants():
                 KokuTenantMiddleware.tenant_cache.clear()
         for name in data:
             LOG.info(f"Deleted tenant: {name}")
+
+
+@celery_app.task(name="masu.processor.tasks.process_openshift_on_cloud", queue=GET_REPORT_FILES_QUEUE, bind=True)
+def process_openshift_on_cloud(self, schema_name, provider_uuid, bill_date, tracing_id=None):
+    """Process OpenShift on Cloud parquet files using Trino."""
+    provider = Provider.objects.get(uuid=provider_uuid)
+    provider_type = provider.type.replace("-local", "")
+
+    table_info = {
+        Provider.PROVIDER_AWS: {"table": "aws_line_items_daily", "date_columns": ["lineitem_usagestartdate"]},
+        Provider.PROVIDER_AZURE: {"table": "azure_line_items", "date_columns": ["usagedatetime", "date"]},
+        Provider.PROVIDER_GCP: {
+            "table": "gcp_line_items_daily",
+            "date_columns": ["usage_start_time", "usage_end_time"],
+        },
+    }
+    table_name = table_info.get(provider_type).get("table")
+    if isinstance(bill_date, str):
+        bill_date = ciso8601.parse_datetime(bill_date)
+    year = bill_date.strftime("%Y")
+    month = bill_date.strftime("%m")
+
+    # We do not have fine grain control over specific days in a month for
+    # OpenShift on Cloud parquet generation. This task will clear and reprocess
+    # the entire billing month passed in as a parameter.
+    where_clause = f" WHERE source='{provider_uuid}' AND year='{year}' AND month='{month}' "
+    table_count_sql = f"SELECT count(*) FROM {table_name}" + where_clause
+
+    count, _ = execute_trino_query(schema_name, table_count_sql)
+    count = count[0][0]
+
+    processor = OCPCloudParquetReportProcessor(
+        schema_name, "", provider_uuid, provider_type, 0, context={"tracing_id": tracing_id, "start_date": bill_date}
+    )
+    remove_files_not_in_set_from_s3_bucket(
+        tracing_id, processor.parquet_ocp_on_cloud_path_s3, 0, processor.error_context
+    )
+    for i, offset in enumerate(range(0, count, settings.PARQUET_PROCESSING_BATCH_SIZE)):
+        query_sql = (
+            f"SELECT * FROM {table_name}"
+            + where_clause
+            + f"OFFSET {offset} LIMIT {settings.PARQUET_PROCESSING_BATCH_SIZE}"
```

Comment:
This can use implicit string concatenation and avoid the (quite minor in this case but significant in other cases) performance hit caused by using the string concatenation operator. I think this makes it slightly more readable, but only slightly. 

```suggestion
            f"SELECT * FROM {table_name}"
            f"{where_clause}"
            f"OFFSET {offset} LIMIT {settings.PARQUET_PROCESSING_BATCH_SIZE}"
```


```
%timeit f"SELECT * FROM {table_name}" + where_clause + f"OFFSET {offset} LIMIT some limit"
200 ns ± 0.706 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

%timeit (f"SELECT * FROM {table_name}" f"{where_clause}" f"OFFSET {offset} LIMIT some limit")
134 ns ± 0.836 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

Reply:
Gladly.